pinot-core/src/main/java/org/apache/pinot/core/common/ColumnValueReader.java

```diff
@@ -0,0 +1,248 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.common;
+
+import org.apache.pinot.core.io.reader.ForwardIndexReader;
+import org.apache.pinot.core.io.reader.ReaderContext;
+import org.apache.pinot.spi.data.FieldSpec.DataType;
+
+
+/**
+ * ColumnValueReader is a wrapper class on top of the forward index reader to read values from it.
```

Comment:
Why do we need this facade over ForwardIndexReader?

Also, this is misleading since ColumnValue implies the actual value in the column whereas if it is passing the call to ForwardIndex, then for dictionary encoded columns you are essentially reading the dictionaryIds which will then be used by DataFetcher to fetch actual values from the dictionary. 

DataFetcher takes care of reading the forward index and if there is dictionary, it then reads the dictionary. So DataFetcher takes care of reading the actual column values in both cases. I don't think we need this. 

DataFetcher should directly use ForwardIndexReader and Dictionary (which it already does)

Having this class also adds to the unclear semantics -- should ForwardIndexReader be enough or should we wrap it around ColumnValueReader? For example, we are using ColumnValueReader in PinotSegmentColumnReader but directly using ForwardIndexReader in IndexSegmentUtils